src/gui/vector/qgsattributeactiondialog.cpp

```diff
@@ -112,6 +112,7 @@ void QgsAttributeActionDialog::insertRow( int row, const QgsAction &action )
   // Type
   item = new QTableWidgetItem( textForType( action.type() ) );
   item->setData( Qt::UserRole, action.type() );
+  item->setData( Qt::UserRole + 1, action.id() );
```

Comment:
could you define a private enum for these two (ActionType, ActionId) ?

Reply:
done